pkg/metric_generator/generator_test.go

```diff
@@ -238,6 +238,95 @@ func TestFilterMetricFamiliesLabels(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "Returns regex allow all labels for metric",
+			allowLabels: allow.Labels(map[string][]string{
+				"node_labels": {
+					".*",
+				},
+			}),
+			familyGenerators: []FamilyGenerator{
+				{
+					Name: "node_labels",
+					Help: "some help",
+					Type: metric.Gauge,
+					GenerateFunc: func(obj interface{}) *metric.Family {
+						return &metric.Family{
+							Metrics: []*metric.Metric{
+								{
+									LabelKeys:   []string{"one", "two", "three"},
+									LabelValues: []string{"value-one", "value-two", "value-three"},
+									Value:       1,
+								},
+							},
+						}
+					},
+				},
+			},
+			results: []FamilyGenerator{
+				{
+					Name: "node_labels",
+					Help: "some help",
+					Type: metric.Gauge,
+					GenerateFunc: func(obj interface{}) *metric.Family {
+						return &metric.Family{
+							Metrics: []*metric.Metric{
+								{
+									LabelKeys:   []string{"one", "two", "three"},
+									LabelValues: []string{"value-one", "value-two", "value-three"},
+									Value:       1,
+								},
+							},
+						}
+					},
+				},
+			},
+		},
+		{
+			name: "Returns deduplicated metrics even though multiple regex's match",
+			allowLabels: allow.Labels(map[string][]string{
+				"node_labels": {
+					"myprefix.*",
+					".*",
+				},
+			}),
+			familyGenerators: []FamilyGenerator{
+				{
+					Name: "node_labels",
+					Help: "some help",
+					Type: metric.Gauge,
+					GenerateFunc: func(obj interface{}) *metric.Family {
+						return &metric.Family{
+							Metrics: []*metric.Metric{
+								{
+									LabelKeys:   []string{"one", "myprefix-label", "myprefix-label2"},
+									LabelValues: []string{"value-one", "value-two", "value-three"},
+									Value:       1,
+								},
+							},
+						}
+					},
+				},
+			},
+			results: []FamilyGenerator{
+				{
+					Name: "node_labels",
+					Help: "some help",
+					Type: metric.Gauge,
+					GenerateFunc: func(obj interface{}) *metric.Family {
+						return &metric.Family{
+							Metrics: []*metric.Metric{
+								{
+									LabelKeys:   []string{"one", "myprefix-label", "myprefix-label2"},
+									LabelValues: []string{"value-one", "value-two", "value-three"},
+									Value:       1,
+								},
+							},
+						}
+					},
+				},
+			},
+		},
```

Comment:
can we add a test to make sure only specified labels are allowed?  for example, something like this...

                {
                        name: "Returns only specified labels",
                        allowLabels: allow.Labels(map[string][]string{
                                "node_labels": {
                                        "label1",
                                },
                        }),
                        familyGenerators: []FamilyGenerator{
                                {
                                        Name: "node_labels",
                                        Help: "some help",
                                        Type: metric.Gauge,
                                        GenerateFunc: func(obj interface{}) *metric.Family {
                                                return &metric.Family{
                                                        Metrics: []*metric.Metric{
                                                                {
                                                                        LabelKeys:   []string{"label1", "label11", "label111"},
                                                                        LabelValues: []string{"value-one", "value-two", "value-three"},
                                                                        Value:       1,
                                                                },
                                                        },
                                                }
                                        },
                                },
                        },
                        results: []FamilyGenerator{
                                {
                                        Name: "node_labels",
                                        Help: "some help",
                                        Type: metric.Gauge,
                                        GenerateFunc: func(obj interface{}) *metric.Family {
                                                return &metric.Family{
                                                        Metrics: []*metric.Metric{
                                                                {
                                                                        LabelKeys:   []string{"label1"},
                                                                        LabelValues: []string{"value-one"},
                                                                        Value:       1,
                                                                },
                                                        },
                                                }
                                        },
                                },
                        },
                },